test/0.4.24/algorithm.generator.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+#
+# Tests generator for Algorithm.frequent() code. Run it with:
+# python algorithm.generator.py
+# yarn test test/0.4.24/algorithm.test.generated.js
+#
+import copy
+import collections
+import random
+
+
+NUM_TESTS = 100
+MAX_NUM = 5
+CASE = """
+    r = await algorithm.frequentTest(%array%, %quorum%, { from: testUser })
+    assert(r.exists === %exists%)
+    assertBn(r.mode, bn(%result%))
+"""
+
+
+def frequent(nums, quorum):
+    ctr = collections.Counter(nums).most_common()
+    if ctr[0][1] >= quorum and (len(ctr) == 1 or ctr[1][1] != ctr[0][1]):
+        return True, ctr[0][0]
+    return False, 0
+
+
+with open('algorithm.test.template.js') as inp:
+    template = inp.read()
+
+cases = ''
+for arr_len in range(2, 10):
+    positive = 0
+    for i in range(NUM_TESTS):
+        nums = []
+        for j in range(arr_len):
+            nums.append(random.randint(1, MAX_NUM))
+        quorum = random.randint(1, arr_len)
```

Comment:
I suggest making the test reproducible by seeding the RNG and logging the seed.

Reply:
Good point, implemented. Although this is mostly a manual test (I mean, not automatically run in any CI/CD) at least now.